File: pyjl/tests/tests_new/bounds_test.py
if __name__ == "__main__":
    seq: list = [1,2,3,4,5] 
    seq_copy: list = [1,2,3,4,5]
    gap = 1
    for i in range(len(seq) - gap):
        # Assert that the list has not changed
        assert seq[i] == seq_copy[i]

    complex_list = [([1,2],3,4)]
    for ([a1,a2], b, c) in complex_list:
        assert a1 == 1
        assert a2 == 2
        assert b == 3
        assert c == 4

        # Nested Loop test
        arr = []
        a = 1
        for j in range(0,2):
            arr.append(a)
            a += 1
        
        assert arr[0] == 1
        assert arr[1] == 2
    
    # Test limit access
    assert seq[0] == 1 # First value
    assert seq[2] == 3 # Some value in the middle
    assert seq[-1] == 5 # Last Value
    
    x = 1
    assert seq[x] == 2

    print("OK")

File: pyjl/tests/tests_new-py2many/bounds_test.jl
function main()
seq::Vector{Int64} = [1, 2, 3, 4, 5]
seq_copy::Vector{Int64} = [1, 2, 3, 4, 5]
gap = 1
for i in (0:length(seq) - gap - 1)
@assert(seq[i + 1] == seq_copy[i + 1])
end
complex_list = [([1, 2], 3, 4)]
for ((a1, a2), b, c) in complex_list
@assert(a1 == 1)
@assert(a2 == 2)
@assert(b == 3)
@assert(c == 4)
arr = []
a = 1
for j in (0:1)
push!(arr, a);
a += 1
end
@assert(arr[1] == 1)
@assert(arr[2] == 2)
end
@assert(seq[1] == 1)
@assert(seq[3] == 3)
@assert(seq[end] == 5)
x = 1
@assert(seq[x + 1] == 2)
println("OK");
end

main()
